### PR TITLE
Pass-down -user-module-version to compiler if the compiler supports it

### PR DESCRIPTION
**Pull request by nkcsgexi**

### CODE REVIEW

**Comment by nkcsgexi** [comment]
@swift-ci please test

File: Sources/SwiftDriver/Jobs/FrontendJobHelpers.swift [hunk 1]
```diff
@@ -180,6 +180,15 @@ extension Driver {
     try commandLine.appendAllArguments(.Xfrontend, from: &parsedOptions)
     try commandLine.appendAll(.coveragePrefixMap, from: &parsedOptions)
 
+    // Pass down -user-module-version if we are working with a compiler that
+    // supports it.
+    if let ver = parsedOptions.getLastArgument(.userModuleVersion)?.asSingle {
+      if isFrontendArgSupported(.userModuleVersion) {
```
Comment (cltnschlosser) [inline Sources/SwiftDriver/Jobs/FrontendJobHelpers.swift:186]:
```suggestion
    if let ver = parsedOptions.getLastArgument(.userModuleVersion)?.asSingle,
       isFrontendArgSupported(.userModuleVersion) {
```
Can be one if statement

File: Sources/SwiftDriver/Driver/Driver.swift [hunk 2]
```diff
@@ -329,6 +329,20 @@ public struct Driver {
       .appending(component: "Frameworks")
   } ()
 
+  public func isFrontendArgSupported(_ opt: Option) -> Bool {
```
Comment (cltnschlosser) [inline Sources/SwiftDriver/Driver/Driver.swift:332]: Do we want this to be public api?

**Review by cltnschlosser — APPROVED** [review]
Some minor comments, otherwise looks good.

File: Sources/SwiftDriver/Driver/Driver.swift (hunk 2 above)
Comment (nkcsgexi) [reply Sources/SwiftDriver/Driver/Driver.swift:332]: Yeah, I think this is potentially useful for libSwiftDriver clients.

**Comment by nkcsgexi** [comment]
@swift-ci please test

**Comment by nkcsgexi** [comment]
@swift-ci please test

**Pull request merged**